diff --git a/grenade.cabal b/grenade.cabal
index 44a5977..6115d35 100644
--- a/grenade.cabal
+++ b/grenade.cabal
@@ -97,5 +97,5 @@ test-suite test
                      , hmatrix
                      , mtl
                      , text                            == 1.2.*
-                     , QuickCheck                      == 2.7.*
+                     , QuickCheck                      >= 2.7         && < 2.9
                      , quickcheck-instances            == 0.3.*
